### ISSUE TITLE
[a11y]: Modal - aria-labelledby attribute of div with classname 'bx--modal-content' is invalid

### ISSUE THREAD

**Issue by LauKwaiFanHK** [carbon-design-system/carbon#13990]
### Package

@carbon/react, carbon-components-react

### Browser

Chrome, Firefox

### Operating System

MacOS

### Package version

v10

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker

### Description

IBM Equal Access Accessibility Checker reports violation: the ARIA attributes "aria-labelledby" are not valid for the element <div> with classname "bx--modal-content". 

Element location:
   ```
 <div
    aria-labelledby="bx--modal-header__label--modal-1"
    class="bx--modal-content"
    id="bx--modal-body--modal-1">
```

This issue also occurs on the Modal componetn on Carbon-react storybook:
![Screenshot 2023-06-13 at 14 39 51](https://github.com/carbon-design-system/carbon/assets/81176980/6d9a979d-cc8b-49be-8fb0-74035823e86c)

### Reproduction/example

https://codesandbox.io/s/friendly-cache-8hhips?file=/src/index.js

### Steps to reproduce

1. Go to Carbon-react storybook: https://v7-react.carbondesignsystem.com/?path=/story/components-modal--default
2. Scan the Modal default component using IBM Equal Access Accessibility Checker
3. You see the Checker reports a violation:   The ARIA attributes "aria-labelledby" are not valid for the element <div> with implicit ARIA role "generic".

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
